### PR TITLE
rustdoc: Remove `PrimitiveType::{to_url_str, as_str}`

### PR DESCRIPTION
**Pull request by jyn514**
These can easily be rewritten in terms of `as_sym`, and this avoids bugs where the two get out of sync.

I don't expect this to have a perf impact, but I'll start a perf run just in case.

### CODE REVIEW

**Comment by jyn514** [comment]
Instruction count looks like noise. Max rss seems very slightly worse, but it's a noisy benchmark so it's hard to tell.

I don't have a strong preference either way on whether to land this.

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
It's fine, the fact that we have less code to maintain counterbalance the very small perf regression. Thanks!

@bors: r+

File: src/librustdoc/passes/collect_intra_doc_links.rs [hunk 1]
```diff
@@ -390,7 +390,8 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
                         ty::AssocKind::Const => "associatedconstant",
                         ty::AssocKind::Type => "associatedtype",
                     };
-                    let fragment = format!("{}#{}.{}", prim_ty.as_str(), out, item_name);
+                    let fragment =
+                        format!("{}#{}.{}", prim_ty.as_sym().to_string(), out, item_name);
```
Comment (petrochenkov) [inline src/librustdoc/passes/collect_intra_doc_links.rs:394]:
```suggestion
                        format!("{}#{}.{}", prim_ty.as_sym(), out, item_name);
```

**Pull request merged**